mutation.hh

```diff
@@ -267,6 +267,12 @@ std::optional<stop_iteration> consume_clustering_fragments(schema_ptr s, mutatio
     position_in_partition::tri_compare cmp(*s);
 
     while (!stop && (crs_it != crs_end || rts_it != rts_end)) {
+        // Dummy rows are part of the in-memory representation but should be
+        // invisible to reads.
+        if (crs_it != crs_end && crs_it->dummy()) {
+            ++crs_it;
+            continue;
+        }
         bool emit_rt;
         if (crs_it != crs_end && rts_it != rts_end) {
             const auto cmp_res = cmp(rts_it->position(), crs_it->position());
```

Comment:
A more generic solution would be to use `crs_it->position().reversed()` here for reverse schema.

Your solution works too, but it assumes that all !dummy() positions are invariant to the clustering order transformation (can be used in both forward and reverse schema). Currently holds, but again, relying on fewer implicit assumptions is good for code stability.